syntax = "proto3";

package metric;

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

// Request to get metrics between start and end time
message MetricRequest {
    google.protobuf.Timestamp start_time = 1;
    google.protobuf.Timestamp end_time = 2;
}

// A single data point
message DataPoint {
    google.protobuf.Timestamp time = 1;
    oneof value {
        double float_value = 2;
        google.protobuf.Empty none_value = 3; // for NaN values
    }
}

// Response containing a list of data points
message MetricResponse {
    repeated DataPoint data_points = 1;
}

// Service definition
service MetricService {
    rpc GetMetrics (MetricRequest) returns (MetricResponse);
}
